esp-hal/src/peripheral.rs

```diff
@@ -257,6 +298,17 @@ mod peripheral_macros {
         }
     }
 
+    #[doc(hidden)]
+    #[macro_export]
+    macro_rules! into_mapped_ref {
```

Comment:
I'm adding this macro here, instead of in a PR that actually uses it, because I'm introducing the `map_into` method(s) here. Can be moved if it's an issue.